opencog/query/AttentionalFocusCB.cc

```diff
@@ -74,3 +76,101 @@ IncomingSet AttentionalFocusCB::get_incoming_set(Handle h)
 
 	return filtered_set;
 }
+
+void AttentionalFocusCB::validate_clauses(std::set<Handle>& vars,
+                                          std::vector<Handle>& clauses)
+{
+}
+
+void AttentionalFocusCB::perform_search(PatternMatchEngine *pme,
+                                        std::set<Handle> &vars,
+                                        std::vector<Handle> &clauses,
+                                        std::vector<Handle> &negations)
+{
+	// In principle, we could start our search at some node, any node,
```

Comment:
this seems to be a cut-n-paste of the default perform_search() method. It deserves a description of how it differs.  Also, since the difference is minor, perhaps there should be some sort of common code to handle both?  Some flag, or some extra parameter?